File: app/services/dataset_service.py
from typing import List
from app.models.dataset import Dataset
from app.models.data_table import DataTable

class DatasetService:
    @staticmethod
    def get_datasets() -> List[Dataset]:
        return Dataset.get_all()
    
    @staticmethod
    def create_dataset(name: str, description: str, database_path: str) -> Dataset:
        dataset = Dataset(
            name=name,
            description=description,
            database_path=database_path,
        )
        return dataset.save()
    
    @staticmethod
    def get_tables(dataset_id: int) -> List[DataTable]:
        return DataTable.get_by_dataset(dataset_id)
    
    @staticmethod
    def get_table_schema(dataset_id: int, table_name: str) -> dict:
        tables = DataTable.get_by_dataset(dataset_id)
        for table in tables:
            if table.table_name == table_name:
                return table.schema_info
        return {}

File: frontend/src/services/datasetService.ts
import api from './api'
import type { Dataset, DataTable } from '../types'

export const datasetService = {
  getDatasets: async (): Promise<Dataset[]> => {
    const response = await api.get('/datasets')
    return response.data.data
  },

  createDataset: async (data: {
    name: string
    description?: string
    database_name?: string
  }): Promise<Dataset> => {
    const response = await api.post('/datasets', data)
    return response.data.data
  },

  getTables: async (datasetId: number): Promise<DataTable[]> => {
    const response = await api.get(`/datasets/${datasetId}/tables`)
    return response.data.data
  },

  createTable: async (datasetId: number, data: {
    table_name: string
    fields: Array<{
      name: string
      type: string
      primaryKey?: boolean
      notNull?: boolean
      default?: any
    }>
  }): Promise<void> => {
    await api.post(`/datasets/${datasetId}/tables`, data)
  },

  addColumn: async (datasetId: number, tableName: string, data: {
    column_name: string
    column_type: string
    not_null?: boolean
    default_value?: any
  }): Promise<void> => {
    await api.post(`/datasets/${datasetId}/tables/${tableName}/columns`, data)
  },
}

